examples/allocation/zig/README.md

```diff
@@ -0,0 +1,16 @@
+## Zig allocation example
+
+This example shows how to pass strings in and out of a Wasm function defined in Zig, built with `zig build`.
+
+Ex.
+```bash
+$ go run greet.go wazero
+wasm >> Hello, wazero!
+go >> Hello, wazero!
+```
+
+Under the covers, [greet.zig](testdata/greet.zig) does a few things of interest:
```

Comment:
make sure these are up to date please, and any other notes of interest in a paragraph like `Note: `. I think there are a couple surprising things, so whatever's left post review should either be a line comment or in the ad-hoc note thing (thinking about exit on each export function unless that's going to be sorted soon)

Reply:
Sure. Added a note about `unreachable`.
https://github.com/tetratelabs/wazero/pull/741/commits/398dc4f0bf6bc056dd044e52fc4e9e9b8f082573